packages/flutter/test/widgets/platform_view_test.dart

```diff
@@ -612,159 +697,238 @@ void main() {
+        ),
+      );
+
+      expect(
+        viewsController.views,
+        unorderedEquals(<FakeIosPlatformView>[
+          FakeIosPlatformView(currentViewId + 2, 'maps')
+        ]),
+      );
+    });
+
+    testWidgets('Dispose UIView ', (WidgetTester tester) async {
+      final FakeIosPlatformViewsController viewsController = FakeIosPlatformViewsController();
+      viewsController.registerViewType('webview');
+      await tester.pumpWidget(
+        Center(
+          child: SizedBox(
+            width: 200.0,
+            height: 100.0,
+            child: UiKitView(viewType: 'webview', layoutDirection: TextDirection.ltr),
+          ),
+        ),
+      );
 
-    int factoryInvocationCount = 0;
-    final ValueGetter<EagerGestureRecognizer> constructRecognizer = () {
-        factoryInvocationCount += 1;
-        return EagerGestureRecognizer();
-      };
+      await tester.pumpWidget(
+        const Center(
+          child: SizedBox(
+            width: 200.0,
+            height: 100.0,
+          ),
+        ),
+      );
+
+      expect(
+        viewsController.views,
+        isEmpty,
+      );
+    });
+
+    testWidgets('UIView survives widget tree change', (WidgetTester tester) async {
+      final int currentViewId = platformViewsRegistry.getNextPlatformViewId();
+      final FakeIosPlatformViewsController viewsController = FakeIosPlatformViewsController();
+      viewsController.registerViewType('webview');
+      final GlobalKey key = GlobalKey();
+      await tester.pumpWidget(
+        Center(
+          child: SizedBox(
+            width: 200.0,
+            height: 100.0,
+            child: UiKitView(viewType: 'webview', layoutDirection: TextDirection.ltr, key: key),
+          ),
+        ),
+      );
 
-    await tester.pumpWidget(
-      AndroidView(
-        viewType: 'webview',
-        gestureRecognizers: <Factory<OneSequenceGestureRecognizer>>[
-          Factory<EagerGestureRecognizer>(constructRecognizer),
-        ].toSet(),
-        layoutDirection: TextDirection.ltr,
-      ),
-    );
+      await tester.pumpWidget(
+        Center(
+          child: Container(
+            child: SizedBox(
+              width: 200.0,
+              height: 100.0,
+              child: UiKitView(viewType: 'webview', layoutDirection: TextDirection.ltr, key: key),
+            ),
+          ),
+        ),
+      );
 
-    await tester.pumpWidget(
-      AndroidView(
-        viewType: 'webview',
-        hitTestBehavior: PlatformViewHitTestBehavior.translucent,
-        gestureRecognizers: <Factory<OneSequenceGestureRecognizer>>[
-          Factory<EagerGestureRecognizer>(constructRecognizer),
-        ].toSet(),
-        layoutDirection: TextDirection.ltr,
-      ),
-    );
+      expect(
+        viewsController.views,
+        unorderedEquals(<FakeIosPlatformView>[
+          FakeIosPlatformView(currentViewId + 1, 'webview')
+        ]),
+      );
+    });
 
-    expect(factoryInvocationCount, 1);
   });
 }
```

Comment:
consider adding tests that hit the "_controller is null" side of everywhere that you end up testing whether _controller is null (doing so will involve creating a bunch of interesting tests that test edge cases of things changing at various times, disposing early, etc, which is generally valuable)

Reply:
added a 'disposeUIView before creation completed' test case